src/Microsoft.DotNet.Cli.Utils/ScriptExecutor.cs

```diff
@@ -45,24 +45,18 @@ public static Command CreateCommandForScript(Project project, string scriptComma
                 var comSpec = Environment.GetEnvironmentVariable("ComSpec");
                 if (!string.IsNullOrEmpty(comSpec))
                 {
-                    scriptArguments =
-                        new[] { comSpec, "/S", "/C", "\"" }
-                        .Concat(scriptArguments)
-                        .Concat(new[] { "\"" })
-                        .ToArray();
+                    scriptArguments = new[] { comSpec, "/S", "/C", ArgumentEscaper.EscapeArgArray(scriptArguments) };
```

Comment:
Here you should call this EscapeArgArray Separately, then call EscapeArgArrayForCmd on that returned string